aptos-move/block-executor/src/executor.rs

```diff
@@ -260,6 +265,8 @@ where
         }
 
         let num_txns = signature_verified_block.len();
+        resize_speculative_logs(num_txns);
```

Comment:
We need to make sure logs are cleared after every block's execution for this. Why not create a new log for each execution? Can this affect performance?

Reply:
flush actually swaps right now, and clears everything (we could get smarter later but for now seems okay, scheduler also allocates similar vectors for statuses). We have to make sure to clear / flush regardless of the resize, otherwise, the next time could log the same things as well.
The next block size we only know the next time we have a block to execute.